packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -6117,37 +6176,65 @@ export default class Meeting extends StatelessWebexPlugin {
   }
 
   /**
-   * Functionality for when a share is ended.
+   * Functionality for when a share audio is ended.
    * @private
    * @memberof Meeting
    * @param {MediaStream} localShare
    * @returns {undefined}
    */
-  private handleShareTrackEnded = async () => {
+  private handleShareAudioTrackEnded = async () => {
     if (this.wirelessShare) {
       this.leave({reason: MEETING_REMOVED_REASON.USER_ENDED_SHARE_STREAMS});
     } else {
       try {
-        await this.unpublishTracks([this.mediaProperties.shareTrack]); // todo: screen share audio (SPARK-399690)
+        await this.unpublishTracks([this.mediaProperties.shareAudioTrack]);
       } catch (error) {
         LoggerProxy.logger.log(
-          'Meeting:index#handleShareTrackEnded --> Error stopping share: ',
+          'Meeting:index#handleShareAudioTrackEnded --> Error stopping share: ',
           error
         );
       }
     }
+    this.triggerStoppedSharing();
+  };
 
-    Trigger.trigger(
-      this,
-      {
-        file: 'meeting/index',
-        function: 'handleShareTrackEnded',
-      },
-      EVENT_TRIGGERS.MEETING_STOPPED_SHARING_LOCAL,
-      {
-        reason: SHARE_STOPPED_REASON.TRACK_ENDED,
+  /**
+   * Functionality for when a share video is ended.
+   * @private
+   * @memberof Meeting
+   * @param {MediaStream} localShare
+   * @returns {undefined}
+   */
+  private handleShareVideoTrackEnded = async () => {
+    if (this.wirelessShare) {
+      this.leave({reason: MEETING_REMOVED_REASON.USER_ENDED_SHARE_STREAMS});
```

Comment:
This is incorrect. When a share with audio is stopped via browser UI, we will get handleShareVideoTrackEnded() and handleShareAudioTrackEnded() called one after another, so we will call this.leave() on the first one and we should be calling it on the 2nd call (only if both audio and video share tracks ended)

Reply:
moved this to triggerStoppedSharing() which I'm calling from both handlers, it checks for both share tracks to have ended